src/dsp_tools/commands/xmlupload/models/deserialise/xmlresource.py

```diff
@@ -112,10 +128,17 @@ def _get_properties(
                     ungrouped_properties.append(XMLProperty.from_node(subnode, "interval", default_ontology))
                 case "hasTitle" | "hasComment" | "hasDescription" | "hasKeyword":
                     ungrouped_properties.append(XMLProperty.from_node(subnode, "text", default_ontology))
-                case _:
+                case str() as x if x.endswith("-prop") and x.startswith(COMPOSITE_PROPS):
```

Comment:
wouldn't it be more straight forward if `COMPOSITE_PROPS` contained `boolean-prop` etc., and you'd just check `str() as x if x in COMPOSITE_PROPS`?

Reply:
Indeed.